### PR TITLE
Added feature to purchase team packages

### PR DESCRIPTION
**Pull request by r0man1337**
As part of this task, I have implemented:

1. NumberInput component
2. Some new computed properties and methods (also changed few old properties)
3. Saving new parameters to the route query
4. Table with price ranges, plan selector and total price calculator

### "Purchase for a team" link
null

### Team package information
first

### Next-steps instructions
second

### CODE REVIEW

File: components/NumberInput.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <v-text-field
+    :value="value"
+    @input="$emit('input', $event)"
+    type="number"
+    :min="min || null"
+    :max="max || null"
+    :label="label"
+    append-outer-icon="mdi-plus-box"
+    @click:append-outer="increment"
```
Comment (jeffreybiles) [inline components/NumberInput.vue:10]: Cool!  I hadn't used these much before, very handy.

File: components/NumberInput.vue [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <v-text-field
+    :value="value"
+    @input="$emit('input', $event)"
+    type="number"
+    :min="min || null"
+    :max="max || null"
```
Comment (jeffreybiles) [inline components/NumberInput.vue:7]: The default value for these is already null.  Is there a reason why we need the `|| null` here as well?

File: pages/account/next-steps.vue [hunk 3]
```diff
@@ -69,6 +75,7 @@
     data(){
       return {
         plan: getPlanWithDefault(this.$auth.user.plan_id),
+        seats: this.$route.query.seats || 1
```
Comment (jeffreybiles) [inline pages/account/next-steps.vue:78]: Once we get the backend set up, we'll be getting this from the user.  Until then, this is a good placeholder.

File: pages/account/next-steps.vue [hunk 4]
```diff
@@ -14,9 +14,15 @@
         <p>If there's not a time on the calendar that works for you, message me and we'll figure something out.</p>
       </NextStepCompletionCard>
     </div>
-
+    <div v-if="seats > 1">
+      <NextStepCompletionCard :markComplete="() => { markComplete('sendTeamMembersInfo')}"
+                              :stepCompleted="next_steps_taken.sendTeamMembersInfo">
+        <p>Please send a list of other team members you want on the plan to <a href="mailto:jeffrey@vuescreencasts.com">jeffrey@vuescreencasts.com</a>.</p>
```
Comment (jeffreybiles) [inline pages/account/next-steps.vue:20]:
```suggestion
        <p>Please send <a href="mailto:jeffrey@vuescreencasts.com">jeffrey@vuescreencasts.com</a> the names and emails of the other {{ seats - 1 }} team team members you want on the plan.</p>
```

File: pages/account/next-steps.vue [hunk 5]
```diff
@@ -14,9 +14,15 @@
         <p>If there's not a time on the calendar that works for you, message me and we'll figure something out.</p>
       </NextStepCompletionCard>
     </div>
-
+    <div v-if="seats > 1">
+      <NextStepCompletionCard :markComplete="() => { markComplete('sendTeamMembersInfo')}"
```
Comment (jeffreybiles) [inline pages/account/next-steps.vue:18]: 👍

File: pages/order.vue [hunk 6]
```diff
@@ -166,4 +259,4 @@
     cursor: none;
     pointer-events: none;
   }
-</style>
```
Comment (jeffreybiles) [inline pages/order.vue:169]: The code is good here (I'll nitpick some things, but... overall, quite good.)  

The file is getting a bit long, however.

I have a couple abstractions in mind.
* Install the Composition API library and put the price calculation into a reactive object in a composition function (my specific technique here might be a bit unusual (Gregg didn't recognize it), and I haven't screencasted about it, so I may end up doing this one myself).
* Putting the pricing table into a single-use component
* Making a reusable radio button component that abstracts the Vuetify component further

Do those component abstractions sound like a good idea to you?  And are there any others you would suggest?

File: pages/order.vue [hunk 7]
```diff
@@ -128,6 +213,14 @@
           ...this.$route.query,
           planTerm: newTerm
         }})
+      },
+      setSeats(newSeats) {
+        this.seats = newSeats
+
+        this.$router.push({path: '/order', query: {
+            ...this.$route.query,
```
Comment (jeffreybiles) [inline pages/order.vue:221]: 👍

File: pages/order.vue [hunk 8]
```diff
@@ -98,28 +147,64 @@
...
       yearlyPrice(){
-        return this.plan['year'].currentPrice
+        return this.plan['year'].prices[this.currentUsersRangeKey]
       },
       monthlyPrice(){
-        return this.plan['month'].currentPrice
+        return this.plan['month'].prices[this.currentUsersRangeKey]
+      },
+      teamPurchaseMaxSavings() {
+        let teamCostDifference = Math.floor(this.plan[this.planTerm].prices["5+"] / this.plan[this.planTerm].prices["1"]  * 100)
```
Comment (jeffreybiles) [inline pages/order.vue:162]: I think we could do a `prices` computed property defined as follows:

```
return this.plan[this.planTerm].prices
```

The pattern is used 4 times already, and would make teamSavings and teamPurchaseMaxSavings more legible.

File: pages/order.vue [hunk 9]
```diff
@@ -5,21 +5,66 @@
...
+                    <td><v-icon>{{getIcon(range)}}</v-icon> {{ range }}</td>
+                    <td>{{ plan['month'].prices[range] | currency }}</td>
+                    <td>{{ plan['year'].prices[range] | currency }}</td>
+                  </tr>
+                  </tbody>
+                </template>
+              </v-simple-table>
+              <NumberInput label="Please select the number of seats:" :min="1" :value="seats" @input="setSeats" />
+              <v-radio-group :value="planTerm" @change="$event => setTerm($event)">
+                <template v-slot:label>
```
Comment (jeffreybiles) [inline pages/order.vue:50]: I prefer using the shortcut `#label` vs `v-slot:label`.

File: pages/order.vue [hunk 10]
```diff
@@ -5,21 +5,66 @@
...
+                </v-radio>
+                <v-radio value="year">
+                  <template v-slot:label>
+                    <div>Yearly</div>
+                  </template>
+                </v-radio>
+              </v-radio-group>
+              <div>Per seat price: {{ currentPrice | currency }}</div>
+              <div>Total price: {{ totalPrice | currency }}</div>
+              <strong>You're saving {{ teamSavings }}% with a team package!</strong>
```
Comment (jeffreybiles) [inline pages/order.vue:66]: An oversight of my original spec: I didn't combine teamSavings with yearlySavings.  We should put that information somewhere, in a way that's not confusing.  Any ideas?

File: pages/order.vue [hunk 11]
```diff
@@ -98,28 +147,64 @@
...
     },
+    created() {
+      let { seats, team } = this.$route.query
+      if (seats) {
+        this.seats = parseInt(seats)
+      }
+      this.isTeamPackageInterfaceEnabled = this.seats > 1 || team === "true"
+    },
     methods: {
+      getIcon(range) {
```
Comment (jeffreybiles) [inline pages/order.vue:190]: Really nice touch with the icons

File: pages/order.vue [hunk 12]
```diff
@@ -5,21 +5,66 @@
...
+                  <tbody>
+                  <tr v-for="range in usersRangeKeys" :key="range">
+                    <td><v-icon>{{getIcon(range)}}</v-icon> {{ range }}</td>
+                    <td>{{ plan['month'].prices[range] | currency }}</td>
+                    <td>{{ plan['year'].prices[range] | currency }}</td>
+                  </tr>
+                  </tbody>
+                </template>
+              </v-simple-table>
+              <NumberInput label="Please select the number of seats:" :min="1" :value="seats" @input="setSeats" />
```
Comment (jeffreybiles) [inline pages/order.vue:48]: Good abstraction

**Comment by jeffreybiles** [comment]
Good work!

A couple things that need to happen before we merge:

On my end:

* Build the backend portion
* Create and connect all the Stripe plans
* Try the Composition API refactor I mentioned
* Some minor cleanup (wording, etc.) where it's less time for me to just change it than to write it out.

On your end:

* Do the small changes requested in comments
* Create more abstractions (along the same lines as NumberInput).  Some suggestions in this comment: https://github.com/jeffreybiles/vue-screencasts-nuxt/pull/9#discussion_r406537119

I don't think any of those items are dependent on the others, so we can work in parallel.

File: pages/order.vue (hunk 6 above)
Comment (jeffreybiles) [reply pages/order.vue:169]: Update: Could you make the radio buttons into a button group instead?  https://vuetifyjs.com/en/components/button-groups/

**Comment by jeffreybiles** [comment]
I'm launching this feature.  Please perform the cleanup/refactors in another branch.

File: pages/order.vue (hunk 9 above)
Comment (r0man1337) [reply pages/order.vue:60]: Funny, I didn't know about that syntax. Thank you.

**Pull request merged**